sdk/common/src/main/java/io/opentelemetry/sdk/internal/GlobUtil.java

```diff
@@ -79,4 +65,42 @@ private static Pattern toRegexPattern(String globPattern) {
     }
     return Pattern.compile(patternBuilder.toString());
   }
+
+  private static class GlobPatternPredicate implements Predicate<String> {
+    private final String globPattern;
+    @Nullable private final Pattern pattern;
+
+    private GlobPatternPredicate(String globPattern) {
+      this.globPattern = globPattern;
+      // If globPattern contains '*' or '?', convert it to a regex and return corresponding
+      // predicate
+      Pattern pattern = null;
+      for (int i = 0; i < globPattern.length(); i++) {
+        char c = globPattern.charAt(i);
+        if (c == '*' || c == '?') {
+          pattern = toRegexPattern(globPattern);
+          break;
+        }
+      }
+      this.pattern = pattern;
+    }
```

Comment:
nit: prefer not to do any real work in a constructor -- would prefer this to take a glob and pattern, and then use a create/factory method instead...and that could live in the containing class if you want.